packages/apollo-server-types/src/index.ts

```diff
@@ -93,7 +93,7 @@ export interface GraphQLRequestContext<TContext = Record<string, any>> {
 
 export type ValidationRule = (context: ValidationContext) => ASTVisitor;
 
-export class InvalidGraphQLRequestError extends Error {}
+export class InvalidGraphQLRequestError extends GraphQLError {}
```

Comment:
This changes the parent class of `InvaidGraphQLRequestError` from `Error` to `GraphQLError` in order to allow such an error to be received by `didEncounterErrors`, which requires `GraphQLError`s.

The `GraphQLError` itself is still an `instanceof Error`, so any existing code that leverages such a condition should still work as expected, and I suspect that most other conditions that could be checked on either an `Error` or `GraphQLError` should also still remain the same.  Despite any uncertain here, I'd call this a net-win for reporting errors more reliably to the hook on which they're expected to be received.